.gitattributes

```diff
@@ -1 +1,5 @@
 * text=auto eol=lf
+/*/**/Dockerfile            linguist-generated
+/*/**/docker-entrypoint.sh  linguist-generated
+/Dockerfile*.template        linguist-language=Dockerfile
```

Comment:
The changes look great overall and will help reviewers focus on the essential parts of the code.

One minor thing: the last line change about .template files seems unrelated to the PR title and description. Perhaps we could tweak the commit message and PR title and description slightly to fully encompass this change as well.

Other than that, I have no concerns. Let me know what you think!

Reply:
A second thought: sometimes, we may still check the generated files to ensure effective changes. Would that be affected if we want to do so? Like, be difficult?

Reply:
It just hides it by default but you can still reveal it (a bit like a file that has too many diff). This would be helpful for bot generated PR.